base/deprecated.jl

```diff
@@ -1348,6 +1348,18 @@ end
 @deprecate srand(filename::AbstractString, n::Integer=4) srand(read!(filename, Array{UInt32}(Int(n))))
 @deprecate MersenneTwister(filename::AbstractString)  srand(MersenneTwister(0), read!(filename, Array{UInt32}(Int(4))))
 
+# when this deprecation is deleted, remove all calls to it, and all `remote=nothing` keyword
+# arguments, and rename `remote_arg` to `remote` in base/libgit2/remote.jl
+@eval LibGit2 function deprecate_remote_keyword(f, sig, remote_arg, remote_kwarg)
+    if remote_kwarg === nothing
+        return remote_arg
+    else
+        msg = "$f($sig; remote=\"$remote_kwarg\") is deprecated, use $f($sig, \"$remote_kwarg\")"
```

Comment:
I wrote that one, and went through a couple refactors of it. it does say instead in it.

the argument order differs from the libgit2 c functions', but I guess that's okay. positionals always run the risk of "what order is it supposed to be" when writing or reading code that uses the method, so I'm neutral on this part of the change